src/backend/access/aocs/aocsam_handler.c

```diff
@@ -1364,8 +1364,8 @@ aoco_index_build_range_scan(Relation heapRelation,
 
 	/* Appendoptimized catalog tables are not supported. */
 	Assert(!is_system_catalog);
-	/* Appendoptimized tables have no data on master. */
-	if (IS_QUERY_DISPATCHER())
+	/* Appendoptimized tables have no data on master unless we are in singlenode mode. */
+	if (IS_QUERY_DISPATCHER() && !IS_UTILITY_OR_SINGLENODE(Gp_role))
```

Comment:
what's diff of IS_QUERY_DISPATCHER and GP_ROLE_DISPATCH ?

Reply:
If connect to dispatcher node of the cluster with utility mode, you'll still get `IS_QUERY_DISPATCHER() == true`.
It only check this query is happened on which node, not the role.